File: 기초 연습/동적계획법/피보나치 수열.c
#include <stdio.h>

int memo[100] = {0, 1};

int f(int n) {
  if (n <= 1)
    return n;
  if (memo[n] == 0)
    memo[n] = f(n-1);
  return memo[n] + memo[n-1];
}

void print(int n) {
  printf("[");
  int i = 0;
  while (i < n) {
    printf("%d, ", memo[i]);
    i++;
  }
  printf("%d]\n", memo[n]);
}

int main() {
  int n;
  scanf("%d", &n);
  printf("%d\n", f(n));
  print(n);
  return 0;
}

File: 기초 연습/동적계획법/피보나치수열.py
memo = [0, 1]

def f(i):
  if i <= 1:
    return i
  if i >= len(memo):
    memo.append(f(i-1) + f(i-2))
  return memo[i]

n = int(input())
print(f(n))
print(memo)
